downloads/swan-lake-release-notes/swan-lake-beta3/RELEASE_NOTE.md

```diff
@@ -0,0 +1,768 @@
+- Introduced the support for external import paths in the gRPC command using the `--proto_path` flag
+- Added the PROTO file name as a suffix for the `ROOT_DESCRIPTOR` constant and `getDescriptorMap` function to fix the re-declared symbol issue when multiple stub files are put into the same module. If you are going to regenerate the stub files with the Swan Lake Beta3 release, you need to change the service annotation like below.
+```ballerina
+@grpc:ServiceDescriptor {
+   descriptor: ROOT_DESCRIPTOR_<PROTO_FILE_NAME>,
+   descMap: getDescriptorMap<Proto_File_Name>()
+}
+service "HelloWorld" on new grpc:Listener(9090) {
+```
+
+##### HTTP Package
+- Enabled HTTP trace and access log support
+- Added HATEOAS link support
+- Introduced the `http:Cache` annotation to the resource signature
+- Introduced support for the service-specific media-type subtype prefix
+- Introduced the introspection resource method to get the generated OpenAPI document of the service
+- Added OAuth2 JWT bearer grant type support for client
+
+##### JWT Package
+- Added HMAC signature support for JWT
+  
+##### Log Package
+- Added observability span context values to the log messages when observability is enabled.
+- Introduced a function(`log:setOutputFile`) to write the log output to a file.
+
+##### OAuth2 Package
+- Added JWT bearer grant type support
+    
+##### SQL Package
+- Added support for `queryRow()` in the database connectors. This method allows retrieving a single row as a record, or a single value from the database.
+```ballerina
+record{} queryResult = sqlClient->queryRow(`SELECT * FROM ExTable where row_id = 1`);
+int count = sqlClient->queryRow(`SELECT COUNT(*) FROM ExTable`);
+```
+
+- Introduced `queryConcat()` and `arrayFlattenQuery()` util functions to create a complex query dynamically.
+  - The `queryConcat()` function creates a parameterized query by concatenating a set of parameterized queries.
+    ```ballerina
+    sql:ParameterizedQuery query = `SELECT * FROM students`;
+    sql:ParameterizedQuery query1 = ` WHERE id < ${id} AND age > ${age}`;
+    sql:ParameterizedQuery sqlQuery = sql:queryConcat(query, query1);
+    ```
+  - The `arrayFlattenQuery()` is introduced to make the array flattening easier. It makes the inclusion of 
+    varying array elements into the query easier by flattening the array to return a parameterized query.
+    ```ballerina
+    int[] ids = [1, 2];
+    sql:ParameterizedQuery sqlQuery = sql:queryConcat(`SELECT * FROM DataTable WHERE id IN (`, sql:arrayFlattenQuery(ids), `)`);
+    ```
+
+##### WebSocket Package
+- Added OAuth2 JWT bearer grant type support for the client
+- Introduced retrying for the WebSocket client
+- Introduced the header annotation and query param binding support
+
+##### FTP Package
+- Added SFTP and related security
+- Added the support for an anonymous user
+
+#### Improvements
+
+##### GraphQL Package
+- Validated the `maxQueryDepth` at runtime as opposed to validating it at compile time
```

Comment:
```suggestion
- Validating the `maxQueryDepth` at runtime as opposed to validating it at compile time
```

Reply:
+1 or something like 

Moved `maxQueryDepth` validation from compile-time to runtime.

Reply:
+1 for thisaru's suggestion above

Reply:
Updated as follows:

Moved the `maxQueryDepth` validation from compile-time to runtime.